### PR TITLE
Add client checks status

### PR DESCRIPTION
**Pull request by MancunianSam**
This sets the status to InProgress on the startUpload mutation as we do this just before we start the client metadata extraction.

The addFilesAndMetadata then sets it to Complete or CompleteWithIssues if there are failures.

### CODE REVIEW

**Review by vimleshtna — COMMENTED** [review]
Looks good, just a few minor comments.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala [hunk 1]
```diff
@@ -40,15 +40,16 @@ class ConsignmentService(
         throw ConsignmentStateException(s"Existing consignment upload status is '${uploadStatus.get.value}', so cannot start new upload")
       }
       val now = Timestamp.from(timeSource.now)
-      val consignmentStatusRow = ConsignmentstatusRow(uuidSource.uuid, startUploadInput.consignmentId, "Upload", "InProgress", now)
-      consignmentRepository.addParentFolder(startUploadInput.consignmentId, startUploadInput.parentFolder, consignmentStatusRow)
+      val consignmentStatusUploadRow = ConsignmentstatusRow(uuidSource.uuid, startUploadInput.consignmentId, "Upload", InProgress, now)
+      val consignmentStatusClientChecksRow = ConsignmentstatusRow(uuidSource.uuid, startUploadInput.consignmentId, "ClientChecks", InProgress, now)
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala:44]: It's a good point, I've swapped out the two "Upload" strings and added another variable in for ClientChecks.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala [hunk 2]
```diff
@@ -66,52 +68,68 @@ class FileService(fileRepository: FileRepository,
...
 
+    generateMatchedRows(consignmentId, rows)
+  }
+
+  private def generateMatchedRows(consignmentId: UUID, rows: Future[List[Rows]]): Future[List[FileMatches]] = {
     for {
       rowsWithMatchId <- rows
       _ <- rowsWithMatchId.grouped(fileUploadBatchSize).map(row => fileRepository.addFiles(row.map(_.fileRow), row.flatMap(_.metadataRows))).toList.head
+      consignmentCheckStatus = rowsWithMatchId.find(_.clientChecks != Completed).map(_.clientChecks).getOrElse(Completed)
+      _ <- consignmentStatusService.addConsignmentStatus(ConsignmentStatusInput(consignmentId, "ClientChecks", consignmentCheckStatus))
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala:99]: I've changed this one as well.

**Review by vimleshtna — APPROVED** [review]
Looks good, thanks.

**Pull request merged**